src/hooks/useViolations.ts

```diff
@@ -79,8 +79,27 @@ function useViolations(violations: TransactionViolation[]) {
                     }));
             }
 
+            // missingTag has special logic for policies with dependent tags, because only one violation is returned for all tags
+            // when no tags are present, so the tag name isn't set in the violation data. That's why we add it here
+            if (data?.policyHasDependentTags && currentViolations[0]?.name === CONST.VIOLATIONS.MISSING_TAG && data?.tagListName) {
+                return [
+                    {
+                        ...currentViolations[0],
+                        data: {
+                            ...currentViolations[0].data,
+                            tagName: data?.tagListName,
+                        },
+                    },
+                ];
+            }
+
             // tagOutOfPolicy has special logic because we have to account for multi-level tags and use tagName to find the right tag to put the violation on
-            if (currentViolations[0]?.name === 'tagOutOfPolicy' && data?.tagListName !== undefined && currentViolations[0]?.data?.tagName) {
+            if (currentViolations[0]?.name === CONST.VIOLATIONS.TAG_OUT_OF_POLICY && data?.tagListName !== undefined && currentViolations[0]?.data?.tagName) {
+                return currentViolations.filter((violation) => violation.data?.tagName === data?.tagListName);
+            }
+
+            // allTagLevelsRequired has special logic because we have to account for tags that are already filled
```

Comment:
```suggestion
            // allTagLevelsRequired has special logic because it is returned when one but not all the tags are set,
            // so we need to return the violation for the tag fields without a tag set
```